admin/code/CampaignAdminList.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+
+namespace SilverStripe\Admin;
+
+use SilverStripe\Forms\FormField;
+use SilverStripe\ORM\Versioning\ChangeSet;
+
+/**
+ * Warning: Volatile API as placeholder for standard "GridField"
+ */
+class CampaignAdminList extends FormField
+{
+    protected $schemaDataType = FormField::SCHEMA_DATA_TYPE_CUSTOM;
+
+    protected $schemaComponent = 'GridField';
+
+    public function getSchemaDataDefaults()
+    {
+        $data = parent::getSchemaDataDefaults();
+
+        // Set admin section options
+        $adminURL = AdminRootController::admin_url();
+        $data['data']['recordType'] = CampaignAdmin::config()->get('tree_class');
+
+        // Merge custom endpoints
+        $data['data']['collectionReadEndpoint'] = [
+            "url" => "{$adminURL}campaigns/sets",
```

Comment:
`campaigns` shouldn't be hardcoded here, the URL route can be changed through config